src/System.Windows.Forms/src/System/Windows/Forms/TrackBar.cs

```diff
@@ -181,6 +182,16 @@ protected override CreateParams CreateParams
                 }
 
                 return cp;
+
+                void EnableAutoTicksIfRequired()
+                {
+                    _autoDrawTicks = false;
+                    if (ShouldAutoDrawTicks())
+                     {
+                        _autoDrawTicks = true;
+                        cp.Style |= (int)PInvoke.TBS_AUTOTICKS;
+                     }
+                }
```

Comment:
I think this can be simplified:
1. Call `_autoDrawTicks = ShouldAutoDrawTicks();` at the top of the property.
2. Update this as follows:
    ```suggestion

                void EnableAutoTicksIfRequired()
                {
                     if (_autoDrawTicks)
                     {
                        cp.Style |= (int)PInvoke.TBS_AUTOTICKS;
                     }
                }
    ```

Reply:
where do we set the  `_autoDrawTicks` to true or false ? . If you change it in `ShouldAutoDrawTicks()` would the values not always be the same when comparing ?. e.g. `if (_autoDrawTicks != ShouldAutoDrawTicks())`

Reply:
At the top of the property:
```cs
        protected override CreateParams CreateParams
        {
            get
            {
                _autoDrawTicks = ShouldAutoDrawTicks();

                CreateParams cp = base.CreateParams;
                ...
```

This is the time when the native control is being created.

> ...would the values not always be the same when comparing ?. e.g. `if (_autoDrawTicks != ShouldAutoDrawTicks())`

No, because `ShouldAutoDrawTicks()` can change based on several conditions.